File: app/src/main/java/com/example/barte_000/chopin/OfferListAdapter.java
package com.example.barte_000.chopin;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

/**
 * Created by barte_000 on 19.11.2016.
 */

public class OfferListAdapter extends RecyclerView.Adapter<OfferListAdapter.APIHolder>{
    private List<Offer> offers;

    public OfferListAdapter(List<Offer> offers) {
        this.offers = offers;
    }

    public void clear() {
        offers.clear();
        notifyDataSetChanged();
    }

    public void addAll(List<Offer> offers) {
        this.offers.addAll(offers);
        notifyDataSetChanged();
    }

    public static class APIHolder extends RecyclerView.ViewHolder {

        public APIHolder(View view) {
            super(view);
        }
    }

    @Override
    public APIHolder onCreateViewHolder(ViewGroup parent, int type) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.fragment_offer_list, parent, false);
        APIHolder apiHolder = new APIHolder(view);
        return apiHolder;
    }
    @Override
    public int getItemCount() {
        return Offer.l.size();
    }

    @Override
    public void onBindViewHolder(APIHolder holder, int position) {

    }

}

File: app/src/main/java/com/chopin/chopin/adapters/OfferListAdapter.kt
package com.chopin.chopin.adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.chopin.chopin.R
import com.chopin.chopin.models.Offer

class OfferListAdapter(private val offers: MutableList<Offer>) : RecyclerView.Adapter<OfferListAdapter.APIHolder>() {

    fun clear() {
        offers.clear()
        notifyDataSetChanged()
    }

    fun addAll(offers: List<Offer>) {
        this.offers.addAll(offers)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, type: Int): APIHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.fragment_offer_list, parent, false)
        return APIHolder(view)
    }

    override fun getItemCount(): Int {
        return offers.size
    }

    override fun onBindViewHolder(offerViewHolder: APIHolder, position: Int) {
    }

    class APIHolder(view: View) : RecyclerView.ViewHolder(view)
}
